lib/ci/run_ci.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var FormData = require('form-data');
+
+const {
+  CI_DOMAIN,
+  CI_TYPES,
+  CI_TYPES_KEYS
+} = require('./ci_type_parser');
+
+const CI_CRUMB_URL = `https://${CI_DOMAIN}/crumbIssuer/api/json`;
+const CI_PR_NAME = CI_TYPES.get(CI_TYPES_KEYS.PR).jobName;
+const CI_PR_URL = `https://${CI_DOMAIN}/job/${CI_PR_NAME}/build`;
+
+class RunPRJob {
+  constructor(cli, request, owner, repo, prid) {
+    this.cli = cli;
+    this.request = request;
+    this.owner = owner;
+    this.repo = repo;
+    this.prid = prid;
+  }
+
+  async getCrumb() {
+    try {
+      const { crumb } = await this.request.json(CI_CRUMB_URL);
+      return crumb;
+    } catch (e) {
+      this.cli.error('Failed to get breadcrumbs');
```

Comment:
the cli error output nesting might be a little weird here  🤔 we might instead want to return `false` in the catch block and then down at the callsite do something like
```
this.cli.startSpinner('Validating Jenkins credentials');
const crumb = await this.getCrumb();

if (!crumb) {
  this.cli.stopSpinner('Jenkins credentials invalid');
  return;
}
this.cli.stopSpinner('Jenkins credentials valid');
```